api.go

```diff
@@ -138,31 +138,39 @@ func HandleQueryValidation(c *gin.Context) {
 	absPath, _ := filepath.Abs("../searchrefiner/dictionary/words.txt")
 	keywordDictionary := loader.LoadDictionary(absPath)
 
+	lang = strings.ToLower(lang)
+
 	var fieldsError []string
 
-	if lang == "medline" {
+	if strings.ToLower(lang) == "medline" {
 		scanner := bufio.NewScanner(strings.NewReader(rawQuery))
 		var extractedFields []string
 		for scanner.Scan() {
 			temp := scanner.Text()
 			line := temp[3:]
 			reg := regexp.MustCompile(`\.([^.]+)\.`)
 			rawFields := reg.FindAllStringSubmatch(line, -1)
-			extractedFields = append(extractedFields, rawFields[0][1])
+			if len(rawFields) > 0 {
+				extractedFields = append(extractedFields, rawFields[0][1])
+			}
 		}
-		for _, i := range extractedFields {
-			flag := checker.CheckWord(fieldsDictionary, strings.ToLower(i), 0)
-			if !flag {
-				fieldsError = append(fieldsError, i)
+		if len(extractedFields) > 0 {
+			for _, i := range extractedFields {
+				flag := checker.CheckWord(fieldsDictionary, strings.ToLower(i), 0)
+				if !flag {
+					fieldsError = append(fieldsError, i)
+				}
 			}
 		}
-	} else if lang == "pubmed" {
+	} else if strings.ToLower(lang) == "pubmed" {
 		reg := regexp.MustCompile(`\[([^]]+)\]`)
 		rawFields := reg.FindAllStringSubmatch(rawQuery, -1)
-		for _, i := range rawFields {
-			flag := checker.CheckWord(fieldsDictionary, strings.ToLower(i[1]), 0)
-			if !flag {
-				fieldsError = append(fieldsError, i[1])
+		if len(rawFields) > 0 {
```

Comment:
Same thing with this check, can be removed (see above).

Reply:
Same as above.